### DISCUSSION TITLE
Issue in getting net metrics

### DISCUSSION THREAD

**Discussion by vikrg23** [The-OpenROAD-Project/OpenROAD#5807] [General]
I am trying to get net metrics like resistance, capacitance and coupling cap in OpenROAD. But the APIs are returning 0.0 for all the nets in my design. I am using post-route db for this. Please let me know if there is a solution for this issue.

I have attached a testcase for this issue. After extracting the file contents, go into the testcase directory and source the runme.tcl file in OpenROAD.
[net_metric_testcase.zip](https://github.com/user-attachments/files/17143363/net_metric_testcase.zip)

**Comment by maliberty** [comment]
Your test case consist of only a runme.tcl and no actual files.

**Reply by vikrg23** [reply]
Sorry my bad @maliberty , attaching proper files.
[net_metric_testcase.zip](https://github.com/user-attachments/files/17144295/net_metric_testcase.zip)

**Reply by vikrg23** [reply]
Hi @maliberty , any fixes for this issue?

**Comment by maliberty** [comment]
read_spef reads the parasitics directly into sta and not into odb.  Trying to query odb fails as nothing is stored there.

**Reply by vikrg23** [reply]
Hi @maliberty  , so what is the correct way to get the parasitic from a design?
